src/gas/GasFeeController.ts

```diff
@@ -0,0 +1,360 @@
+import type { Patch } from 'immer';
+
+import EthQuery from 'eth-query';
+import { v1 as random } from 'uuid';
+import { BaseController } from '../BaseControllerV2';
+import { safelyExecute } from '../util';
+import type { RestrictedControllerMessenger } from '../ControllerMessenger';
+import type {
+  NetworkController,
+  NetworkState,
+} from '../network/NetworkController';
+import {
+  fetchGasEstimates as defaultFetchGasEstimates,
+  fetchLegacyGasPriceEstimate as defaultFetchLegacyGasPriceEstimate,
+  calculateTimeEstimate,
+} from './gas-util';
+
+export type unknownString = 'unknown';
+
+export interface EstimatedGasFeeTimeBounds {
+  lowerTimeBound: number | null;
+  upperTimeBound: number | unknownString;
+}
+
+/**
+ * @type LegacyGasPriceEstimate
+ *
+ * A single gas price estimate for networks and accounts that don't support EIP-1559
+ *
+ * @property gasPrice - A GWEI hex number, the result of a call to eth_gasPrice
+ */
+
+export interface LegacyGasPriceEstimate {
+  gasPrice: string;
+}
+
+/**
+ * @type Eip1559GasFee
+ *
+ * Data necessary to provide an estimate of a gas fee with a specific tip
+ *
+ * @property minWaitTimeEstimate - The fastest the transaction will take, in milliseconds
+ * @property maxWaitTimeEstimate - The slowest the transaction will take, in milliseconds
+ * @property suggestedMaxPriorityFeePerGas - A suggested "tip", a GWEI hex number
+ * @property suggestedMaxFeePerGas - A suggested max fee, the most a user will pay. a GWEI hex number
+ */
+
+interface Eip1559GasFee {
```

Comment:
We need to export this interface since customGas values is going to be a partial of this.